main.js

```diff
@@ -42,4 +42,407 @@ for (let i = 0; i < array3.length; i++) {
+    },
+    {
+      name: 'Shilla Stay',
+      city: 'Seoul',
+      country: 'South Korea',
+    },
+    {
+      name: 'San Firenze Suites',
+      city: 'Florence',
+      country: 'Italy',
+    },
+    {
+      name: 'The Andaman Resort',
+      city: 'Port Dickson',
+      country: 'Malaysia',
+    },
+    {
+      name: 'Black Penny Villas',
+      city: 'BTDC, Nuca Dua',
+      country: 'Indonesia',
+    },
+    {
+      name: 'Koko Hotel',
+      city: 'Tokyo',
+      country: 'Japan',
+    },
+    {
+      name: 'Ramada Plaza',
+      city: 'Istanbul',
+      country: 'Turkey',
+    },
+    {
+      name: 'Waikiki Resort Hotel',
+      city: 'Hawaii',
+      country: 'USA',
+    },
+    {
+      name: 'Puro Hotel',
+      city: 'Krakow',
+      country: 'Poland',
+    },
+    {
+      name: 'Asma Suites',
+      city: 'Santorini',
+      country: 'Greece',
+    },
+    {
+      name: 'Cityden Apartments',
+      city: 'Amsterdam',
+      country: 'Netherlands',
+    },
+    {
+      name: 'Mandarin Oriental',
+      city: 'Miami',
+      country: 'USA',
+    },
+    {
+      name: 'Concept Terrace Hotel',
+      city: 'Rome',
+      country: 'Italy',
+    },
+    {
+      name: 'Ponta Mar Hotel',
+      city: 'Fortaleza',
+      country: 'Brazil',
+    },
+    {
+      name: 'Four Seasons Hotel',
+      city: 'Sydney',
+      country: 'Australia',
+    },
+    {
+      name: 'Le Meridien',
+      city: 'Nice',
+      country: 'France',
+    },
+    {
+      name: 'Apart Neptun',
+      city: 'Gdansk',
+      country: 'Poland',
+    },
+    {
+      name: 'Lux Isla',
+      city: 'Ibiza',
+      country: 'Spain',
+    },
+    {
+      name: 'Nox Hostel',
+      city: 'London',
+      country: 'UK',
+    },
+    {
+      name: 'Leonardo Vienna',
+      city: 'Vienna',
+      country: 'Austria',
+    },
+    {
+      name: 'Adagio Aparthotel',
+      city: 'Edinburgh',
+      country: 'UK',
+    },
+    {
+      name: 'Steigenberger Hotel',
+      city: 'Hamburg',
+      country: 'Germany',
+    },
+  ];
+
+function search (str1) {
+      return hotels 
+      .filter((input) => (input.name + input.city + input.country).includes(str1))
+      .map(input => `${input.name}, ${input.city}, ${input.country}`);
+  }
+  console.log(search("Hotel"));
+
+/* Сопоставте страны с городами из массива:
+дан массив;
+напишите функцию, которая выберет все уникальные страны и сопоставит с ними города;
+в консоли должен быть выведен примерно такой результат:
+{
+   Australia: ['Sydney'],
+   Germany: ['Berlin', 'Hamburg'],
+   Italy: ['Florence', 'Rome'],
+   USA: ['Chicago', 'Hawaii', 'Miami'],
+   Ukraine: ['Kyiv']
+} */
+let countries = [];
+// hotels.forEach(element => console.log(element.country))
+let countryCity = {};
+hotels.forEach(element => countries.push(element.country))
+countries = [...new Set (countries)]
+countries.forEach(land => {
+    countryCity[land]=[];
+    Object.keys(hotels).forEach(element => {
```

Comment:
for the future, its better to call variables according to their role or to what it really seems